src/browser-detector/BrowserDetector.d.ts

```diff
@@ -0,0 +1,12 @@
+import { PureComponent } from 'react';
+
+type Browser = 'chrome' | 'firefox' | 'safari' | 'msie' | 'msedge' | 'opera';
+
+export type BrowserDetectorProps = {
+  children: any, // TODO: t.ReactChildren
+  placeholder: (detectedBrowser: string) => any, // TODO: t.ReactChildren
```

Comment:
woops... these are the real types, and they're more interesting :P https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/bowser/index.d.ts